File: gdbm/jython/delete/gdbm_delete.py
#! /usr/bin/jython
# -*- coding: utf-8 -*-
#
#	gdbm_delete.py
#
#					Oct/07/2010
import	sys
import	string
#import anydbm
import gnudbm
#
sys.path.append ('/var/www/uchida/data_base/common/python_common')
#
#from dbm_manipulate import dbm_disp_proc,dbm_delete_proc
# -------------------------------------------------------------
print ("*** 開始 ***")
#
id_in = string.atoi (sys.argv[1])
print ("%d" % id_in)
#
db_name = "/var/tmp/gdbm/cities.pag";
#dd = anydbm.open (db_name,"c")
dd = gdbm.open (db_name,"c")
#
#dbm_delete_proc	(dd,id_in)
key_id = str (id_id)
del dd[key_id]
#
#dbm_disp_proc (dd)
#
dd.close ()
#
print ("*** 終了 ***")
# -------------------------------------------------------------

File: gdbm/jruby/delete/gdbm_delete.rb
#! /usr/bin/ruby
# -*- encoding: utf-8 -*-
#
#	delete/gdbm_delete.rb
#
#				Aug/18/2010
#
require	'dbm'
require	'date'
require	'json'
#
load '/var/www/uchida/data_base/common/ruby_common/kvalue_manipulate.rb'
# --------------------------------------------------------------------
puts	"*** 開始 ***"
#
id_in = ARGV[0].to_i
#
puts id_in
#
ee=DBM.open("/var/tmp/gdbm/cities")
#
kvalue_delete_proc(ee,id_in)
#
ww=ee.to_hash;
kvalue_display_proc(ww)
ee.close
#
print "ww.size =",ww.size,"\n"
#
puts "*** 終了 ***"
#
